http/client.go

```diff
@@ -14,46 +14,92 @@ import (
 	"time"
 )
 
-// New creates a new http.Client with a custom certificate, which can be loaded from the passed CA Bundle file and/or
-// directory. If both CABundleFile and CABundleDir are empty arguments, it creates an unsecure HTTP client.
-func New(CABundleFile, CABundleDir string, httpTimeout time.Duration) (*http.Client, error) {
-	return _new(CABundleFile, CABundleDir, httpTimeout, "")
-}
+type ClientOption func(*http.Client) error
 
-// NewAcceptInvalidHostname new http.Client with ability to accept HTTPS certificates that don't
-// match the hostname of the server they are connecting to.
-func NewAcceptInvalidHostname(CABundleFile, CABundleDir string, httpTimeout time.Duration, hostname string) (*http.Client, error) {
-	return _new(CABundleFile, CABundleDir, httpTimeout, hostname)
-}
+// New creates a new http.Client with the Passed Client Options
+// that will have a custom certificate if it is loaded from the passed CA Bundle file and/or
+// directory. If both CABundleFile and CABundleDir are empty arguments, it creates an unsecure HTTP client.
+func New(opts ...ClientOption) (*http.Client, error) {
+	t := &http.Transport{
+		TLSClientConfig: &tls.Config{},
+	}
 
-func _new(CABundleFile, CABundleDir string, httpTimeout time.Duration, acceptInvalidHostname string) (*http.Client, error) {
-	// go default http transport settings
-	t := &http.Transport{}
+	client := &http.Client{
+		Transport: t,
+	}
 
-	if CABundleFile != "" || CABundleDir != "" {
-		certs, err := getCertPool(CABundleFile, CABundleDir)
-		if err != nil {
+	for _, opt := range opts {
+		if err := opt(client); err != nil {
 			return nil, err
 		}
+	}
+
+	return client, nil
+}
+
+// WithCABundleFile adds the CABundleFile cert to the the client's certPool,
+// if a CABundleDir is passed it will be joined to the CABundleFile to try to get the file.
+// If the file is a full path, even it the CABundleDir is passed, it will be detected and the full
+// path will be used instead
+func WithCABundleFile(CABundleFile, CABundleDir string) ClientOption {
```

Comment:
Not much a fan of this behavior, I'd prefer to just supply the path to the file directly.

Reply:
I know, but I don't want to break it for users who are using it, what do you think? this is a behaviour that is actually documented on the Unit Test, so it was a feature at the time.

Reply:
Done